### ISSUE TITLE
Add support for rapid api and mashape-key generation

### ISSUE THREAD

**Issue by KenEucker** [KenEucker/imgur#113]
**Is your feature request related to a problem? Please describe.**
We could add support for using Rapid API requests over standard api.imgur.com requests.

**Describe the solution you'd like**
A rapid API host and key could be added to the Credentials of this project in order to support RapidAPI. These are the fields to add:
```ts
'x-rapidapi-host': 'imgur-apiv3.p.rapidapi.com',
'x-rapidapi-key': 'RAPID_API_KEY'
```

**Describe alternatives you've considered**
Alternatively, there doesn't seem to be a way for commercial users of this API to be able to set these fields. Another library would need to be chosen or this library forked.

**Additional context**
We should support this, since it is the defacto solution provided by [Imgur's documentation](https://apidocs.imgur.com/#:~:text=API.%20Happy%20hacking!-,Commercial%20Usage,-Your%20application%20is): 

> To use Imgur's API commercially, you must first register your application. Once that's done, you must register with RapidAPI. RapidAPI allows you to choose a pricing plan that fits your needs. From then on, the API endpoint is https://imgur-apiv3.p.rapidapi.com/ which must be used in place of https://api.imgur.com/. Additionally, you must set a X-Mashape-Key request header with the key obtained from RapidAPI.

**Comment by cacheflow** [comment]
Seems straightforward. Seems like it's just adding some headers and changing the endpoint if they're using the RapidAPI endpoint, right?

**Comment by KenEucker** [comment]
@cacheflow adding headers and then allowing for the baseURL being used here:
https://github.com/KenEucker/imgur/blob/main/src/client.ts#L50

To be set to the RapidAPI host, based on the configuration.

**Comment by KenEucker** [comment]
#122 includes updates that implement this feature.

**Comment by KenEucker** [comment]
This feature was published with imgur@2.1.0
